helm/resource_release.go

```diff
@@ -1384,7 +1387,9 @@ func chartPathOptions(d resourceGetter, m *Meta, cpo *action.ChartPathOptions) (
 	cpo.Keyring = d.Get("keyring").(string)
 	cpo.RepoURL = repositoryURL
 	cpo.Verify = d.Get("verify").(bool)
-	cpo.Version = version
+	if !isLocalChart(chartName, cpo.RepoURL) || registry.IsOCI(repository) {
```

Comment:
I just tried with this config:

```terraform
resource "helm_release" "nginx_ingress" {
  name       = "redis"

  chart      = "oci://registry-1.docker.io/bitnamicharts/redis"

  version = "17.14.2"

  set {
    name = "architecture"
    value = "standalone"
  }
}
```

And it still installs the latest version instead of the one I've specified in `.version`. Your check here only checks if `repository` is set to an OCI registry but you can specify a full OCI URL in `.chart`.